enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,273 @@
+
+In the beginning the goal is to support 2 profiles:
+
+- `full` (same as today)
+- `operational` (only collect metrics necessary for alerts, recording rules, telemetry and dashboards)
+
+When the cluster admin enables the `operational` profile, the k8s Prometheus resource would be configured accordingly:
+
+```yaml
+apiVersion: monitoring.coreos.com/v1
+kind: Prometheus
+metadata:
+  name: k8s
+  namespace: openshift-monitoring
+spec:
+  serviceMonitorSelector:
+    matchExpressions:
+    - key: monitoring.openshift.io/scrape-profile
+      operator: NotIn
+      values:
+      - "full"
+```
+ 
+An OpenShift team that would want to support the scrape profiles feature would need to provide 2 monitors for each profile (in this example 1 service monitor per profile).
+
+```yaml
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    monitoring.openshift.io/scrape-profile: full
+  name: foo-full
+  namespace: openshift-bar
+spec:
+  endpoints:
+    port: metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: foo
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    monitoring.openshift.io/scrape-profile: operational
+  name: foo-operational
+  namespace: openshift-bar
+spec:
+  endpoints:
+    port: metrics
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: foo
+  metricRelabelings:
+  - sourceLabels: [__name__]
+    action: keep
+    regex: "requests_total|requests_failed_total"
+ ```
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+  // Defines the scraping profile that will be enforced on the platform
+  // prometheus instance. Possible values are full and operational.
+  //
+  // +kubebuilder:validation:Enum=full;operational
+  // +optional
+	ScrapeProfile string `json:"scrapeProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints [optional]
+
+Each OpenShift team that wants to adopt this feature will be responsible for providing the different monitors. This work is not trivial, dependecies beween operators exist, this makes unclear for developers if it's really fine or not for their operator in the "operational" profile to not expose a given metric. To try and help teams with this problem we would provide a tool that would consume a monitor and generate the "operational" monitor for that operator. However for this, the tool would need to have access to an up to date, instalation of OpenShift in order to query the Prometheus instance running in the cluster.
```

Comment:
```suggestion
Each OpenShift team that wants to adopt this feature will be responsible for providing the different monitors. This work is not trivial. Dependencies between operators and their metrics exist. This makes it difficult for developers to determine whether a given metric must be added to the "operational" profile or not. To aid teams with this problem we will provide a tool that would consume a monitor and generate the "operational" monitor for that operator. However for this, the tool would need to have access to an up to date, installation of OpenShift in order to query the Prometheus instance running in the cluster.
```

Reply:
Much better! I was struggling with this one 😅